compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -520,12 +550,12 @@ impl<'a> Parser<'a> {
         }
 
         if self.check_too_many_raw_str_terminators(&mut err) {
-            if expected.contains(&TokenType::Token(token::Semi)) && self.eat(&token::Semi) {
+            return if expected.contains(&TokenType::Token(token::Semi)) && self.eat(&token::Semi) {
                 err.emit();
-                return Ok(true);
+                Ok(true)
```

Comment:
Can you change this back? I think the old way was clearer.

Reply:
Done.